dd4t-caching/src/main/java/org/dd4t/caching/providers/EHCache3Provider.java

```diff
@@ -159,6 +159,7 @@ public void invalidate(final String key) {
             LOG.error("Detected undeclared null payload on element with key " + key + " at insert time!",
                     exToLogToHaveStacktraceWhoCausedIt);
             cacheElement.setNull(true);
+            //@ToDo probably bug!
```

Comment:
take a look at this place. we have set up expiration flag and then immediately after it reset

Reply:
@RaimondKempees please take a look at this, it sets up the value which immediately changed